--- a/ecm_interface.c
+++ b/ecm_interface.c
@@ -8863,7 +8863,6 @@ static struct ctl_table ecm_interface_ta
 		.proc_handler		= &ecm_interface_igs_enabled_handler,
 	},
 #endif
-	{ }
 };
 
 #ifdef ECM_INTERFACE_IPSEC_GLUE_LAYER_SUPPORT_ENABLE
--- a/frontends/ecm_front_end_common.c
+++ b/frontends/ecm_front_end_common.c
@@ -1237,7 +1237,6 @@ static struct ctl_table ecm_front_end_sy
 		.mode		= 0644,
 		.proc_handler	= &ecm_front_end_tcp_denied_ports_handler,
 	},
-	{}
 };
 
 /*
--- a/frontends/sfe/ecm_sfe_common.c
+++ b/frontends/sfe/ecm_sfe_common.c
@@ -467,7 +467,6 @@ static struct ctl_table ecm_sfe_sysctl_t
 		.proc_handler   = &ecm_sfe_mht_enable_handler,
 	},
 #endif
-	{}
 };
 
 /*
